modules/utxo-bin/src/commands.ts

```diff
@@ -32,16 +32,115 @@ import {
   getRange,
   parseIndexRange,
 } from './generateAddress';
+import { convertTransaction } from './convertTransaction';
 
 type OutputFormat = 'tree' | 'json';
 
-type ArgsParseTransaction = {
+type ArgsReadTransaction = {
   network: string;
   stdin: boolean;
-  clipboard: boolean;
   path?: string;
-  txid?: string;
   data?: string;
+  clipboard: boolean;
+  txid?: string;
+  cache: boolean;
+  finalize: boolean;
+};
+
+function addReadTransactionOptions<T>(b: yargs.Argv<T>): yargs.Argv<T & ArgsReadTransaction> {
+  return b
+    .option('network', { alias: 'n', type: 'string', demandOption: true })
+    .option('stdin', { type: 'boolean', default: false })
+    .option('path', { type: 'string', nargs: 1, default: '' })
+    .option('data', { type: 'string', description: 'transaction bytes (hex or base64)', alias: 'hex' })
+    .option('clipboard', { type: 'boolean', default: false })
+    .option('txid', { type: 'string' })
+    .option('cache', {
+      type: 'boolean',
+      default: false,
+      description: 'use local cache for http responses',
+    })
+    .option('finalize', {
+      type: 'boolean',
+      default: false,
+      description: 'finalize PSBT and parse result instead of PSBT',
+    });
+}
+
+async function readTransaction(argv: ArgsReadTransaction, httpClient: HttpClient): Promise<ParserTx> {
+  const network = getNetworkForName(argv.network);
+  let data;
+
+  if (argv.txid) {
+    data = await fetchTransactionHex(httpClient, argv.txid, network);
+  }
+
+  if (argv.stdin || argv.path === '-') {
+    if (data) {
+      throw new Error(`conflicting arguments`);
+    }
+    console.log('Reading from stdin. Please paste hex-encoded transaction data.');
+    console.log('After inserting data, press Ctrl-D to finish. Press Ctrl-C to cancel.');
+    if (process.stdin.isTTY) {
+      data = await readStdin();
+    } else {
+      data = await fs.promises.readFile('/dev/stdin', 'utf8');
+    }
+  }
+
+  if (argv.clipboard) {
+    if (data) {
+      throw new Error(`conflicting arguments`);
+    }
+    data = await clipboardy.read();
+  }
+
+  if (argv.path) {
+    if (data) {
+      throw new Error(`conflicting arguments`);
+    }
+    data = (await fs.promises.readFile(argv.path, 'utf8')).toString();
+  }
+
+  if (argv.data) {
+    if (data) {
+      throw new Error(`conflicting arguments`);
+    }
+    data = argv.data;
+  }
+
+  // strip whitespace
+  if (!data) {
+    throw new Error(`no txdata`);
+  }
```

Comment:
Possible to move all this in a separate function `validateArguments` or such? This whole function will become easier to reason about then, it's quite long currently.

Reply:
split it up into two methods